### PR TITLE
Make empty string searches be consistent with case (in)sensitivity

### LINKED ISSUE #108968: Term matching on empty string behaves differently when case_insensitive==true
### Elasticsearch Version

8.15.0

### Installed Plugins

_No response_

### Java Version

_bundled_

### OS Version

Elastic Cloud

### Problem Description

Context in [Discuss forum post](https://discuss.elastic.co/t/term-query-with-empty-string-and-case-insensitive-setting-become-invalid/359961). 

The `term` query on a keyword field with an empty value ("") behaves differently depending on the `case_insensitive` flag:
- If `case_insensitive==false` (default), the query matches on "",
- If `case_insensitive==true`, the query does not match on "".

We should make this consistent.

### Steps to Reproduce

Tested on an Elastic Cloud deployment with version 8.15.0-SNAPSHOT.

```
PUT term-match-test
{
  "mappings": {
    "properties": {
      "text": {
        "type": "keyword"
      }
    }
  }
}
```

```
PUT term-match-test/_doc/1
{
  "text": ""
}
```

Case sensitive search returns 1 hit:
```
POST term-match-test/_search
{
  "query": {
    "term": {
      "text": {
        "value": ""
      }
    }
  }
}
```

Case insensitive search returns 0 hits:
```
POST term-match-test/_search
{
  "query": {
    "term": {
      "text": {
        "value": "",
        "case_insensitive": true
      }
    }
  }
}
```

### Logs (if relevant)

_No response_

### PR DESCRIPTION
**Pull request by benwtrent**
If we determine that the searchable term is completely empty, we switch back to a regular term query. This way we return the same docs as expected when we do a case sensitive search.

closes: https://github.com/elastic/elasticsearch/issues/108968

### CODE REVIEW

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search/171_term_query.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+          index: index_with_blank_keyword
+          body: {"query" : {"term" : {"keyword_field" : {"value":  ""}}}}
+
+  - match: { hits.total: 1 }
+
+  - do:
+      search:
+        rest_total_hits_as_int: true
+        index: index_with_blank_keyword
+        body: { "query": { "term": { "keyword_field": {"value": "", "case_insensitive": true } } } }
```
Comment (carlosdelest) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search/171_term_query.yml:35]: Don't we check for the number of hits on this query?

File: docs/changelog/110833.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+pr: 110833
+summary: Make empty string searches be consistent with case (in)sensitivity
+area: Search
+type: bug
+issues: []
```
Comment (carlosdelest) [inline docs/changelog/110833.yaml:5]:
```suggestion
issues: [108968]
```

**Review by carlosdelest — APPROVED** [review]
LGTM - a question on test assertions that may be needed

**Review by tteofili — APPROVED** [review]
LGTM

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Pull request merged**